admin/spec/back-end/service/pupil-access-arrangements.service.spec.js

```diff
@@ -159,4 +160,26 @@ describe('pupilAccessArrangementsService', () => {
       )
     })
   })
+  describe('deletePupilAccessArrangements', () => {
+    it('returns pupil data when successfully deleting relevant access arrangements', async () => {
+      spyOn(pupilDataService, 'sqlFindOneBySlugAndSchool').and.returnValue({id: 1, foreName: 'foreName', lastName: 'lastName'})
+      spyOn(pupilAccessArrangementsDataService, 'sqlDeletePupilsAccessArrangements')
+      const pupilData = await pupilAccessArrangementsService.deletePupilAccessArrangements('urlSlug', 9991001)
+      expect(pupilData).toEqual({id: 1, foreName: 'foreName', lastName: 'lastName'})
+      expect(pupilDataService.sqlFindOneBySlugAndSchool).toHaveBeenCalled()
+      expect(pupilAccessArrangementsDataService.sqlDeletePupilsAccessArrangements).toHaveBeenCalled()
+    })
+    it('rejects if url slug is not present', async () => {
+      spyOn(pupilDataService, 'sqlFindOneBySlugAndSchool')
+      spyOn(pupilAccessArrangementsDataService, 'sqlDeletePupilsAccessArrangements')
+      try {
+        await pupilAccessArrangementsService.deletePupilAccessArrangements()
+        fail()
```

Comment:
We should fail with a message - sorry if I wasn't clear.

Reply:
@jon-shipley missed your message, done.